dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -124,15 +124,9 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
   }
 
   @GetMapping
-  public @ResponseBody List<IconResponse> getAllIcons(
-      @RequestParam(required = false) String[] keywords) {
-    List<Icon> icons;
-
-    if (keywords == null) {
-      icons = iconService.getIcons();
-    } else {
-      icons = iconService.getIcons(keywords);
-    }
+  public @ResponseBody List<IconResponse> getAllIcons(IconCriteria iconCriteria) {
+
+    List<? extends Icon> icons = iconService.getIcons(iconCriteria);
```

Comment:
```suggestion
    List<Icon> icons = iconService.getIcons(iconCriteria);
```

or even better inline it in the return statement